skins/laika/src/store/create_data_persister.ts

```diff
@@ -0,0 +1,111 @@
+import { MutationPayload, Store } from 'vuex';
+import { DataPersistenceRepository } from '@/view_models/DataPersistenceRepository';
+import { DataPersistenceItem, DataPersistenceMutationType } from '@/view_models/DataPersistenceItem';
+import { DataEncryptor, SubtleDataEncryptor } from '@/view_models/DataEncryptor';
+
+export class DataPersister {
+	dataEncryptor: DataEncryptor;
+	repository: DataPersistenceRepository;
+	keyNamespace: string;
+	initialValues: { key: string, value: any }[];
+
+	constructor( dataEncryptor: DataEncryptor, repository: DataPersistenceRepository, keyNamespace: string ) {
+		this.dataEncryptor = dataEncryptor;
+		this.repository = repository;
+		this.keyNamespace = keyNamespace;
+		this.initialValues = [];
+	}
+
+	async save( key: string, data: string ) {
+		const encrypted = await this.dataEncryptor.encrypt( data );
+		if ( encrypted !== undefined ) {
+			this.repository.setItem( `${this.keyNamespace}/${key}`, encrypted );
+		}
+	}
+
+	async load( key: string ) {
+		const data = this.repository.getItem( `${this.keyNamespace}/${key}` );
+
+		if ( !data ) {
+			return null;
+		}
+
+		try {
+			return await this.dataEncryptor.decrypt( data );
+		} catch ( e ) {
+			this.repository.removeItem( `${this.keyNamespace}/${key}` );
+			return null;
+		}
+	}
+
+	getPlugin( items: DataPersistenceItem[] ) {
+		return ( store: Store<any> ) => {
+			store.subscribe( ( mutation: MutationPayload, state: any ) => {
+				const item = items.find( item => item.mutation === mutation.type );
```

Comment:
This re-use of `item` both as const and function param had me confused for a moment. Maybe change one of the names?